tests/adapter/dbt/tests/adapter/simple_snapshot/test_snapshot.py

```diff
@@ -0,0 +1,203 @@
+
+    def test_updates_are_captured_by_snapshot(self, project):
+        """
+        Update the last 5 records. Show that all ids are current, but the last 5 reflect updates.
+        """
+        self.update_fact_records(
+            {"updated_at": "updated_at + interval '1 day'"}, "id between 16 and 20"
+        )
+        run_dbt(["snapshot"])
+        self._assert_results(
+            ids_with_current_snapshot_records=range(1, 21),
+            ids_with_closed_out_snapshot_records=range(16, 21),
+        )
+
+    def test_inserts_are_captured_by_snapshot(self, project):
+        """
+        Insert 10 records. Show that there are 30 records in `snapshot`, all of which are current.
+        """
+        self.insert_fact_records("id between 21 and 30")
+        run_dbt(["snapshot"])
+        self._assert_results(
+            ids_with_current_snapshot_records=range(1, 31), ids_with_closed_out_snapshot_records=[]
+        )
+
+    def test_deletes_are_captured_by_snapshot(self, project):
+        """
+        Hard delete the last five records. Show that there are now only 15 current records and 5 expired records.
+        """
+        self.delete_fact_records("id between 16 and 20")
+        run_dbt(["snapshot"])
+        self._assert_results(
+            ids_with_current_snapshot_records=range(1, 16),
+            ids_with_closed_out_snapshot_records=range(16, 21),
+        )
+
+    def test_revives_are_captured_by_snapshot(self, project):
+        """
+        Delete the last five records and run snapshot to collect that information, then revive 3 of those records.
+        Show that there are now 18 current records and 5 expired records.
+        """
+        self.delete_fact_records("id between 16 and 20")
+        run_dbt(["snapshot"])
+        self.insert_fact_records("id between 16 and 18")
+        run_dbt(["snapshot"])
+        self._assert_results(
+            ids_with_current_snapshot_records=range(1, 19),
+            ids_with_closed_out_snapshot_records=range(16, 21),
+        )
+
+    def test_new_column_captured_by_snapshot(self, project):
+        """
+        Add a column to `fact` and populate the last 10 records with a non-null value.
+        Show that all ids are current, but the last 10 reflect updates and the first 10 don't
+        i.e. if the column is added, but not updated, the record doesn't reflect that it's updated
+        """
+        self.add_fact_column("full_name", "varchar(200) default null")
+        self.update_fact_records(
+            {
+                "full_name": "first_name || ' ' || last_name",
+                "updated_at": "updated_at + interval '1 day'",
+            },
+            "id between 11 and 20",
+        )
+        run_dbt(["snapshot"])
+        self._assert_results(
+            ids_with_current_snapshot_records=range(1, 21),
+            ids_with_closed_out_snapshot_records=range(11, 21),
+        )
+
+
+class TestSnapshotCheck(SimpleSnapshotTestBase):
```

Comment:
To make it accessible to downstream repos, we're gonna need to strip off the `Test` from these names and redefined them with the pass keyword
[Example](https://github.com/dbt-labs/dbt-core/blob/a5131ecc7d6450ffdc3abe44a75df33c7b1b41e7/tests/adapter/dbt/tests/adapter/dbt_debug/test_dbt_debug.py#L67) -- if you don't do it this way, you'll end up re-running tests from the `import`

Reply:
Upon looking, I noticed that in that example, the other tests are similarly have their own checks that aren't in the Base class. The one you linked is only relying on the check_project and has no tests of its own, which isn't the case here.
To change the class name here and add `pass` means to move the `test_` methods all into the base.
Let me know if I understood